### DISCUSSION TITLE
Custom diffusivity function: interpolation of the current

### DISCUSSION THREAD

**Discussion by mj621** [pybamm-team/PyBaMM#2391] [Q&A]
Hello everybody,

I am trying to implement a custom diffusivity function, which calculates the diffusivity by interpolation. But using the current raises a NotImplementedError. Is there a workaround on how to get the value from "Current function [A]"?

```

from pybamm import Interpolant
current_interpolant = pybamm.Interpolant(drive_cycle[:, 1], drive_cycle[:, 0], timescale * pybamm.t)
param["Current function [A]"] = current_interpolant


def graphite_diffusivity(sto, T):
    current = pybamm.LithiumIonParameters().dimensional_current_with_time
    current_arr= [1, 4]
    diff_coeff_arr= [6.9908116e-17, 1.16713149e-15]

    solution = Interpolant(current, current_arr, diff_coeff_arr)
    return solution
```

`method self.evaluate() not implemented for symbol Current function [A] of type <class 'pybamm.expression_tree.parameter.FunctionParameter'>`

Thanks!

**Comment by valentinsulzer** [comment]
Allowing this change would require quite a big change to the code base. I can tell you how to do it but it's not an easy change

**Reply by mj621** [reply]
I thought so, but I would like to know what the changes would be. Thank you!
